src/xmpp/iq.c

```diff
@@ -2654,6 +2655,37 @@ _mam_rsm_id_handler(xmpp_stanza_t* const stanza, void* const userdata)
     return 0;
 }
 
+void
+iq_register_new_account(const char* const user, const char* const password)
+{
+    xmpp_ctx_t* const ctx = connection_get_ctx();
+    xmpp_stanza_t* iq = stanza_register_new_account(ctx, user, password);
+
+    const char* id = xmpp_stanza_get_id(iq);
+    iq_id_handler_add(id, _register_new_account_result_id_handler, NULL, NULL); // FIXME: function doesn't seem to ever run?
```

Comment:
You have `iq_register_new_account()` outcommented in `cmd_register()` so the function can't run (if I interpret your comment right here).

Reply:
Yeah, that was intentional. That comment in the function meant that the `_register_new_account_result_id_handler` function was seemingly never running, indicating it wasn't properly getting the response. I commented out the `iq_register_new_account` to make way for the new stuff I did with the `_register_handler` and all that, which actually registers after connecting like it should.

Left the `iq_register_new_account` there in case the current implementation can be adapted to the standard iq function thing.

Plus, I've verified that it does actually properly register. the current problems that need fixing are those 2 segfaults.